packages/frontend-v2/components/Sidebar/Sidebar.tsx

```diff
@@ -60,163 +61,174 @@ const createCourseMap = (
-  const concentration = major?.concentrations.concentrationOptions.find(
-    (concentration) => concentration.title === selectedPlan.concentration
-  );
+const Sidebar: React.FC<SidebarProps> = memo(
+  ({ selectedPlan, transferCourses }) => {
+    const router = useRouter();
+    const {
+      major,
+      isLoading: isMajorLoading,
+      error,
+    } = useMajor(selectedPlan.catalogYear, selectedPlan.major);
+    const concentration = major?.concentrations.concentrationOptions.find(
+      (concentration) => concentration.title === selectedPlan.concentration
+    );
 
-  const workerRef = useRef<Worker>();
+    const workerRef = useRef<Worker>();
 
-  const [validationStatus, setValidationStatus] = useState<
-    MajorValidationResult | undefined
-  >(undefined);
+    const [validationStatus, setValidationStatus] = useState<
+      MajorValidationResult | undefined
+    >(undefined);
 
-  const revalidateMajor = () => {
-    setValidationStatus(undefined);
-    if (!selectedPlan || !major || !workerRef.current) return;
+    const revalidateMajor = () => {
+      setValidationStatus(undefined);
+      if (!selectedPlan || !major || !workerRef.current) return;
 
-    currentRequestNum += 1;
-    const validationInfo: WorkerPostInfo = {
-      major: major,
-      taken: getAllCoursesFromPlan(selectedPlan),
-      concentration: selectedPlan.concentration,
-      requestNumber: currentRequestNum,
-    };
-    workerRef.current?.postMessage(validationInfo);
-  };
-
-  // Set up the web worker to handle major validation for us. This helps keep the
-  // UI thread free to display our app, preventing UI freezes while our schedule
-  // is being validated.
-  useEffect(() => {
-    if (!workerRef.current) {
-      workerRef.current = new Worker(
-        new URL("../../validation-worker/worker.ts", import.meta.url)
-      );
-      workerRef.current.onmessage = (message: MessageEvent<WorkerMessage>) => {
-        switch (message.data.type) {
-          case WorkerMessageType.Loaded:
-            revalidateMajor();
-            break;
-          case WorkerMessageType.ValidationResult:
-            // Only update valdation information if it was from the latest request.
-            // This helps us avoid displaying outdated information that could be sent
-            // due to race conditions.
-            if (message.data.requestNumber === currentRequestNum) {
-              setValidationStatus(message.data.result);
-            }
-
-            break;
-          default:
-            throw new Error("Invalid worker message!");
-        }
+      currentRequestNum += 1;
+      const coursesTaken = [
+        ...getAllCoursesFromPlan(selectedPlan),
+        ...transferCourses,
+      ];
+      const validationInfo: WorkerPostInfo = {
+        major: major,
+        taken: coursesTaken,
+        concentration: selectedPlan.concentration,
+        requestNumber: currentRequestNum,
       };
```

Comment:
this was the only change in this file, everything else was prettier doing its thing. essentially just added the transfer courses to the "coursesTaken" array passed to the validation algorithm.